src/vs/workbench/contrib/terminal/electron-browser/terminalInstance.ts

```diff
@@ -1063,12 +1093,24 @@ export class TerminalInstance implements ITerminalInstance {
 
 		// Set the new shell launch config
 		this._shellLaunchConfig = shell; // Must be done before calling _createProcess()
-		// Initialize new process
-		this._createProcess();
+
+		// Initialize new process if we have one.
+		if (this._shellLaunchConfig.executable) {
```

Comment:
Pretty sure this will break a lot of things, executable can be null in which case the default is used

Reply:
Correct. Fixed.